packages/core/ios/Sources/RNDdSdkConfiguration.swift

```diff
@@ -53,3 +54,46 @@ extension NSDictionary {
         )
     }
 }
+
+extension NSArray {
+    /*
+     * Adapts the data format from the React Native SDK configuration to match with the
+     * iOS SDK configuration. For example:
+     *
+     * RN config: [{ match: "example.com", propagatorType: [DATADOG, B3] }]
+     * iOS config: { "example.com": [DATADOG, B3] }
+     */
+    func asFirstPartyHosts() -> [String: Set<TracingHeaderType>] {
+        var firstPartyHosts: [String: Set<TracingHeaderType>] = [:]
+        
+        forEach { h in
```

Comment:
you could use `return .reduce(...)` instead. Something like:

```swift
return reduce(into: [:], { partialResult, h in
   var host = (h as? NSDictionary)

    if let match = (host?.value(forKey: "match") as? String),
       let propagatorTypes = (host?.value(forKey: "propagatorTypes") as? NSArray) {
        partialResult[match] = propagatorTypes.asTracingHeaderType()
    }
})
```

Haven't tested this code.